posthog/models/organization.py

```diff
@@ -68,6 +69,12 @@ def bootstrap(
                 user.current_organization = organization
                 user.current_team = team
                 user.save()
+            if EE_AVAILABLE and AvailableFeature.ROLE_BASED_ACCESS in organization.available_features:
```

Comment:
Q: Are organisations only created by using `bootstrap`, or is there some place where they're called by directly calling `create`?

Reply:
It looks like it's mostly "officially" created through `bootstrap` except for tests and internal metrics stuff? Maybe @Twixes knows more here too

Reply:
Not sure where `User.bootstrap` is called, but this might be problematic:

https://github.com/PostHog/posthog/blob/master/posthog/models/user.py#L70

Other that this, yeah, looks like it's just tests, or internal metrics.

Reply:
ok I see it looks like that's when users sign up for the first time completely

Reply:
Hmm, I don't quite see why we need a default role. It seems cumbersome to ensure all orgs have a default role and that every person belongs to that role, when we could just implicitly defer to the organization/project's access settings. In my mind, a role would usually mean a real-world team (e.g. "Experimentation") or a role within a company ("Project Manager"), so a vague "Write" role just doesn't sit well with me… Is there some more context for this architectural choice?

Reply:
@liyiy 's comment:

How would you differentiate a full edit access vs view only access for a regular team member then? We can rename the role if that's the concern, I just used the example name from launchdarkly's "default roles". But how do you think we would determine "default full edit access" for everyone?

---

I quite like this idea, tbh. Let default permission come from org settings (which determines what the default is - full access or view only), which are overridden by any roles if found for the given user.

Reply:
okay, so by default being a member+ will grant you edit access. And then if you want to restrict certain members to view only, then you add a "view only" membership?